### PR TITLE
BUG: Allow ``None`` as ``api_version`` in ``__array_namespace__`` method

### PR DESCRIPTION
**Pull request by mtsokol**
Hi @rgommers,

This bugfix PR updates `ndarray.__array_namespace__` implementation so: 
```python
x.__array_namespace__(api_version=None)
```
is now allowed, as stated in [the standard](https://data-apis.org/array-api/latest/API_specification/generated/array_api.array.__array_namespace__.html#array_api.array.__array_namespace__).

### CODE REVIEW

File: numpy/_core/src/multiarray/methods.c [hunk 1]
```diff
@@ -2760,19 +2760,28 @@ static PyObject *
...
-        PyErr_Format(PyExc_ValueError,
-                     "Version \"%s\" of the Array API Standard is not supported.",
-                     array_api_version);
-        return NULL;
+    if (array_api_version != Py_None) {
+        if (!PyUnicode_Check(array_api_version))
+        {
+            PyErr_Format(PyExc_ValueError,
+                "Only None and strings are allowed as the Array API version.",
+                array_api_version);
```
Comment (rgommers) [inline numpy/_core/src/multiarray/methods.c:2775]: minor: this looks like a leftover. you can either remove it, or add something like `", but received ..."`.

**Review by rgommers — APPROVED** [review]
Thanks @mtsokol. LGTM module one minor thing

File: numpy/_core/src/multiarray/methods.c (hunk 1 above)
Comment (mtsokol) [reply numpy/_core/src/multiarray/methods.c]: Right, I added `array_api_version` to the error message.

**Review by rgommers — APPROVED** [review]
Thanks Mateusz!

**Pull request merged**